lib/fibonacci.rb

```diff
@@ -1,8 +1,18 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)  (should be O(n))
 # Hint, you may want a recursive helper method
+FIB_NUMS = { 0 => 0, 1 => 1}
+
 def fibonacci(n)
```

Comment:
This works, but you're maintaining the entire sequence of fibonacci numbers.  However you only need to keep the last 2 fib numbers.  Can you think to reduce the amount of space used?